src/main/scala/centaur/test/Test.scala

```diff
@@ -134,13 +140,25 @@ object Operations {
     }
   }
 
+  def retrieveMetadata(workflow: SubmittedWorkflow): Test[WorkflowMetadata] = {
+    new Test[WorkflowMetadata] {
+      override def run: Try[WorkflowMetadata] = {
+        val response = MetadataRequest(Get(CentaurConfig.cromwellUrl + "/api/workflows/v2/" + workflow.id + "/metadata"))
+        // Try to convert the response to a Metadata in our return Try.
+        // Currently any error msg will be opaque as it's unlikely to be an issue (can flesh out later)
+        sendReceiveFutureCompletion(response map { r => WorkflowMetadata.fromMetadataJson(r).toOption.get })
```

Comment:
I left a comment here, but it's a Won't Fix (i.e. the copypasta was intentional)